### DISCUSSION TITLE
preconfigure for MDM

### DISCUSSION THREAD

**Discussion by ScharrerZie** [rkkr/simple-keyboard#469] [Q&A]
We are deploying this application to our tablets using Mobile Device Management (MDM). Is it possible to preconfigure certain settings by sending an intent or importing a settings file?

**Comment by rkkr** [comment]
No there is no specific code included to allow it and I'll probably not add it. To my limited knowledge about MDM there is no way to import settings into an app. If there is a way then you can find the settings names in https://github.com/rkkr/simple-keyboard/blob/master/app/src/main/java/rkr/simplekeyboard/inputmethod/latin/settings/Settings.java .

There is an adb backup/restore mechanism https://github.com/eviabs/Android-Backup-and-Restore-Guide/blob/master/Android%20Backup%20and%20Restore%20Guide.md which should theoretically work. Don't know if it's useful.

How do you configure other apps? And what do you want to change from default? You can also fork this repo if you want to.

**Comment by ScharrerZie** [comment]
we have some apps, that work with an intent to set some settings. but forking is also an option. 
we set the keyboard height to 75% and turn of sounds and vibrating and the following settings:
PREF_HIDE_SPECIAL_CHARS
PREF_SHOW_NUMBER_ROW
PREF_ENABLED_SUBTYPES
PREF_HIDE_LANGUAGE_SWITCH_KEY

would it be possible to compile a version with these settings predefined?

**Comment by rkkr** [comment]
Just set whatever default values you want in the same `Settings.java` file. It's second param in `prefs.getX(Y, default)`.

**Comment by ScharrerZie** [comment]
Hi, I got a different solution for that, which probably works also for other users with MDMs. Its a common solution for Business use apps an dInhave seen it on many apps like Outlook, Gboard, etc.

see #537 for the solution.
